packages/react-component-library/src/components/NumberInput/NumberInput.tsx

```diff
@@ -2,50 +2,84 @@ import React, { FormEvent } from 'react'
 import classNames from 'classnames'
 import isFinite from 'lodash/isFinite'
 import isNil from 'lodash/isNil'
+import without from 'lodash/without'
 import { v4 as uuidv4 } from 'uuid'
 
 import { EndAdornment } from './EndAdornment'
+import { Footnote } from './Footnote'
 import { Input } from './Input'
 import { StartAdornment } from './StartAdornment'
 import { useFocus } from './useFocus'
 import { useValue } from './useValue'
-import { Footnote } from './Footnote'
+import { UNIT_POSITION } from './constants'
+
+type UnitPosition = typeof UNIT_POSITION.AFTER | typeof UNIT_POSITION.BEFORE
 
 export interface NumberInputProps {
   autoFocus?: boolean
   className?: string
-  isDisabled?: boolean
   footnote?: string
   id?: string
-  isCondensed?: boolean,
+  isCondensed?: boolean
+  isDisabled?: boolean
   label?: string
   max?: number
   min?: number
   name: string
   onBlur?: (event: React.FormEvent) => void
   onChange: (event: any) => void
   placeholder?: string
+  startAdornment?: React.ReactNode | string
   step?: number
+  unit?: string
+  unitPosition?: UnitPosition
   value?: number
-  startAdornment?: React.ReactNode | string
+}
+
+function formatValue(
+  displayValue: number,
+  unit: string,
+  unitPosition: UnitPosition
+) {
+  if (isNil(displayValue)) {
+    return null
+  }
+
+  if (unit) {
+    return unitPosition === UNIT_POSITION.AFTER
+      ? `${displayValue} ${unit}`
+      : `${unit} ${displayValue}`
+  }
+
+  return displayValue
+}
+
+function getNewValue(event: React.FormEvent<HTMLInputElement>, unit: string) {
+  const { value } = event.currentTarget
+  const valueParts = value.split(' ')
+  const valueWithoutUnit = without(valueParts, unit).join()
+
+  return parseInt(valueWithoutUnit, 10)
```

Comment:
Is there ever a possibility this value is sometimes a stepped float? 

Maybe the component doesn't support that currently?

Reply:
The component does not support that currently. I will add something to the backlog as it's a more involved change.